api/capi/src/nnstreamer-capi-util.c

```diff
@@ -933,46 +935,35 @@ ml_validate_model_file (char **model, unsigned int num_models,
 
   switch (*nnfw) {
     case ML_NNFW_TYPE_ANY:
-      /** @todo Make sure num_models is correct */
-      if (g_str_has_suffix (path_down, ".tflite")) {
-        /**
-         * .tflite is supported by both tensorflow and nnfw.
-         * Priority decided with ini file.
-         */
-        gboolean nnfw_runtime_priority =
-            nnsconf_get_custom_value_bool ("nnfw-runtime",
-            "prioritize_tflite_ext", FALSE);
-        gboolean available_nnfw =
-            ml_nnfw_is_available (ML_NNFW_TYPE_NNFW, ML_NNFW_HW_ANY);
-        gboolean available_tflite =
-            ml_nnfw_is_available (ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY);
-
-        if ((nnfw_runtime_priority && available_nnfw) ||
-            (!nnfw_runtime_priority && !available_tflite)) {
-          ml_logi ("The given model [%s] is supposed a nnfw model.", model[0]);
-          *nnfw = ML_NNFW_TYPE_NNFW;
-        } else {
-          ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
-              model[0]);
-          *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
-        }
-      } else if ((g_str_has_suffix (path_down, ".so") &&
-              g_str_has_suffix (path_down_2, ".nb")) ||
-          (g_str_has_suffix (path_down, ".nb") &&
-              g_str_has_suffix (path_down_2, ".so"))) {
+      detected_fw =
+          gst_tensor_filter_framework_auto_detection ((const gchar **) model,
```

Comment:
It looks that gst_tensor_filter_framework_auto_detection returns a g_strdup()ed -gchar. 

Q. Is it freed after all?

Reply:
Right. detected_fw should be released. I will update it.